src/storage/src/hummock/store/state_store.rs

```diff
@@ -299,14 +298,14 @@ impl HummockStorageCore {
         epoch: u64,
         read_options: ReadOptions,
     ) -> StorageResult<HummockStorageIterator> {
-        let compaction_group_id = self.get_compaction_group_id(read_options.table_id).await?;
         // 1. build iterator from staging data
         let (imms, uncommitted_ssts, committed) = {
             let read_guard = self.read_version.read();
-            let (imm_iter, sstable_info_iter) =
-                read_guard
-                    .staging()
-                    .prune_overlap(epoch, compaction_group_id, &key_range);
+            let (imm_iter, sstable_info_iter) = read_guard.staging().prune_overlap(
+                epoch,
+                StaticCompactionGroupId::NewCompactionGroup as CompactionGroupId,
```

Comment:
We can store table id instead of the compaction group id in the shared buffer batch, and here we only need to provide the table id for filtering.

Reply:
This may be related to #5867

Reply:
Since we do not need to split memory data to different group, it is not necessary to store compaction group id in both shared-buffer-batch and sstableinfo

Reply:
> Since we do not need to split memory data to different group, it is not necessary to store compaction group id in both shared-buffer-batch and sstableinfo

see #5867